clippy_lints/src/implicit_saturating_sub.rs

```diff
@@ -0,0 +1,152 @@
+                    if BinOpKind::Gt == cond_op.node || BinOpKind::Ne == cond_op.node {
+                        (cond_left, cond_right)
+                    } else {
+                        return;
+                    }
+                } else if SpanlessEq::new(cx).eq_expr(cond_right, target) {
+                    if BinOpKind::Lt == cond_op.node || BinOpKind::Ne == cond_op.node {
+                        (cond_right, cond_left)
+                    } else {
+                        return;
+                    }
+                } else {
+                    return;
+                };
+
+                // Check if the variable in the condition statement is an integer
+                if !cx.tables.expr_ty(cond_var).is_integral() {
+                    return;
+                }
+
+                // Get the variable name
+                let var_name = ares_path.segments[0].ident.name.as_str();
+
+                match cond_num_val.kind {
+                    ExprKind::Lit(ref cond_lit) => {
+                        // Check if the constant is zero
+                        if let LitKind::Int(0, _) = cond_lit.node {
+                            if cx.tables.expr_ty(cond_left).is_signed() {
+                                return;
+                            } else {
+                                print_lint_and_sugg(cx, &var_name, expr);
+                            }
+                        } else {
+                            return;
+                        }
+                    },
+                    ExprKind::Path(ref cond_num_path) => {
+                        if match_qpath(cond_num_path, &["i8", "MIN"]) || match_qpath(cond_num_path, &["i16", "MIN"]) || match_qpath(cond_num_path, &["i32", "MIN"]) || match_qpath(cond_num_path, &["i64", "MIN"]) {
+                            print_lint_and_sugg(cx, &var_name, expr);
+                        } else {
+                            return;
+                        }
+                    },
+                    ExprKind::Call(ref func, _) => {
+                        if let ExprKind::Path(ref cond_num_path) = func.kind {
+                            if match_qpath(cond_num_path, &["i8", "min_value"]) || match_qpath(cond_num_path, &["i16", "min_value"]) || match_qpath(cond_num_path, &["i32", "min_value"]) || match_qpath(cond_num_path, &["i64", "min_value"]) {
+                                print_lint_and_sugg(cx, &var_name, expr);
+                            } else {
+                                return;
+                            }
+                        } else {
+                            return;
+                        }
+                    },
+                    _ => (),
+                }
+            }
+        }
+    }
+}
+
+fn print_lint_and_sugg(cx: &LateContext<'_, '_>, var_name: &str, expr: &Expr<'_>) {
+    let applicability = Applicability::MaybeIncorrect;
```

Comment:
Any reason for this? I think this could be `MachineApplicable`

Reply:
No reason actually